### ISSUE TITLE
[@types/MathJS] defined as namespace ambient declarations, but needs to depend on module declarations

### ISSUE THREAD

**Issue by andnp** [DefinitelyTyped/DefinitelyTyped#24638]
- [x] I tried using the `@types/mathjs` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
  - Authors: @siavol, @andy-ms, @sandersn, @mhegazy 

Sorry for the "shotgun" tagging of authors here, I tried skimming through previous issues to make sure I was not retreading old ground.

I am using mathjs and a lot of the types are declared as an ambient type declaration. The mathjs lib depends on `Decimal.js` for the `BigNumber` interface within mathjs (which is currently defined as `{}`). `Decimal.js` publishes its own types, but only as module declarations:
```typescript
// decimal.d.ts
export declare class Decimal { ... }
```
```typescript
// mathjs/index.d.ts
declare namespace mathjs {
  ...
  export interface BigNumber {
  }
  ...
}
declare module 'mathjs' {
  export = math;
}
```

Because `BigNumber` is really just a `Decimal` in-disguise, I would like to do:
```typescript
// mathjs/index.d.ts
import { Decimal } from 'decimal.js';
declare namespace mathjs {
  ...
  export interface BigNumber extends Decimal {
    ... // some stuff that BigNumber adds
  }
  ...
}
export = math;
```

This works, except that it will break most (if not all) consumers of `@types/mathjs` because we will go from an ambient declaration to a module declaration.

From the way I see it, my options are:

1) Do it and bump the major version of the types (which I don't know is possible or is against convention)
2) Do it and not care if it breaks people (obviously not that great of an option)
3) Not use the `decimal.js` types, and copy them over manually (also not ideal)
4) Contact the `decimal.js` author and ask him to give an ambient definition that I could pull in with `/// <reference path=... />` (not really ideal for that author to have to yet another types export)

I'm not quite sure which way to move forward with this, or if anyone knows of a better way than I outlined above. I want to avoid breaking people's workflows, but right now the `mathjs` types are incomplete enough to not be usable by me and I'd like to help fix that.

**Comment by sandersn** [comment]
You can use the new(ish) `export as namespace x` statement to maintain backward compatibility with browser-based users that don't use modules. I played around with mathjs and got this:

```ts
import { Decimal } from 'decimal.js';
export as namespace math;
export = mathjs;
declare var mathjs: mathjs.IMathJsStatic; // note that this is now `mathjs`, to merge with the namespace
declare namespace mathjs {
  // .. same code as before
}
// remove the declare module 'mathjs' at the end.
```

This made everything work for me both for module usage and for global (browser-style) usage. Anyway, this should be a pretty easy change to make. Let me know if you need more advice.

**Comment by andnp** [comment]
Yup, this is spot on what I was looking for. I must not have entered the correct combination of search terms :) 

Thanks very much for your help!

**Comment by kaktus40** [comment]
Hello, I have mathjs@3.20.1 and @types/mathjs@3.20.0 installed for my project. With this definition, I have the following error : ``` error TS2307: Cannot find module 'decimal.js'```.  When I comment the line 8 ( ```// import { Decimal } from 'decimal.js';```), I have no difficulty. Is it useful to maintain this problematic line or how should I do?

**Comment by sandersn** [comment]
@kaktus40 this is the kind of question that Stack Overflow is designed for, so if my answer doesn't help, please try re-asking it there.

It sounds like decimal.js isn't installed. Can you try doing `npm install decimal.js`? However, [the npm list of dependencies](https://www.npmjs.com/package/@types/mathjs) shows that the mathjs types depend on decimal.js. So maybe your module setting is incorrect in tsconfig.

**Comment by kaktus40** [comment]
Well, in fact I should have post my question to Stack Overflow. I posted here because my project worked well before I made an update for the latest version of @types/mathjs. Also I checked my installation, changed reference of mathjs to math in my code and I have all the dependencies of mathjs in my node_modules (decimal.js). 
In a word what compiled before doesn't compile now and I don't know why commenting the import makes all work...
